fleet_adapter_r3/fleet_adapter_r3/RobotClientAPI.py

```diff
@@ -531,34 +534,21 @@ def navigate_robot(self, robot_encoding_id: str, time_stamp: float, content: Nav
 
     def navigation_completed(self, robot_name: str) -> NavigationStatus:
         self.refresh_expired_token()
-        
-        with self._lock:
-            mission_status = self.get_mission_status(robot_name=robot_name)
-            robot_status = self.get_robot_status(robot_name=robot_name)
 
-        if mission_status is None or robot_status is None:
-            return NavigationStatus.EMPTY
-        
-        robot_mission_status = mission_status['missionStatus']['mission']['status']
-
-        if robot_mission_status == RobotMissionStatus.MOVING_FINISHED.value:
-            # 5 seconds to wait for any error code
-            # This is necessary as mission status and response codes are from two separate message payloads
-            retries = 5
-            while retries > 0:
-                # within this time window, if any error code comes in, navigation was not completed
-                if ResponseCode.P2P_STARTED not in mission_status['alertIds']:
-                    return NavigationStatus.NAVIGATION_ERROR
-                
-                time.sleep(1)
-                retries -= 1
+        threshold = self.waypoint_buffer_threshold
 
-            # P2P_STARTED indicates moving started and finished successfully without any errors in between
-            return NavigationStatus.NAVIGATION_SUCCESS
-                
+        curr_robot_position = self.position(robot_name)
+
+        if self.target_position is None:
+            print("target_waypoint not set yet. Ignoring...")
+            return False
         else:
-            return NavigationStatus.NAVIGATING if robot_status['status'] == RobotStatus.MOVING.value \
-            else NavigationStatus.NAVIGATION_ERROR
+            # Check if robot is near target waypoint within buffer threshold
+            if (abs(curr_robot_position.x - self.target_position.x) <= threshold and
+                abs(curr_robot_position.y - self.target_position.y) <= threshold):
+                return True
```

Comment:
question: does this mean we don't care about the robot orientation? Concern is just robot is still in the p2p mission but is pending rotation to the target orientation

Reply:
For now, it does not care about robot orientation in order to allow for an lenient criteria for a mission to complete. Depending on the deployment use case, the robot orientation can be taken in account easily.

Reply:
Would it cause an issue though? Since there can be a case where robot is still in the middle of a p2p mission (e.g. still rotating to the target orientation) but the adapter already returns `true`?

Reply:
The robot still completes the p2p mission, despite the adapter returning `true`. 

However, will go back to using the original approach since that has been verified extensively now to work reliably still, contrary to prior findings.